### ISSUE TITLE
Editor: Decrease Line Indentation uses tab-size not spaces-per-indent

### ISSUE THREAD

**Issue by mikewyer** [Komodo/KomodoEdit#1744]
### Short Summary
Decrease Line Indentation removes all the whitespace from the start of the line, when it should only remove one indentation level.

### Steps to Reproduce
Bind a key to Editor: Decrease Line Indentation (Ctl-c, < or shift-TAB or ctrl-shift-TAB)
In python mode (also happens in Python3 mode and perl mode), type out the following snippet:
```
def foo():
  if True:
    print 'bar'
    print 'done'
```
Hit shift-TAB (or whatever) to decrease the indent of the last line

### Expected results
```
def foo():
  if True:
    print 'bar'
  print 'done'
## << 2 spaces above these 
```
### Actual results
```
def foo():
  if True:
    print 'bar'
print 'done'
## ^^ no spaces at all :(
```
### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?*  10.0.1
*Operating System (and version)?* 
Ubuntu 14.04.4 (Trusty) 

### Additional Information

 -- eg. Error logs, screenshots, workarounds --

**Comment by mitchell-as** [comment]
Hi, what are your indentation settings? What happens when you press "Tab" on a new line? Does it insert 2 spaces or 4 spaces?

**Comment by mikewyer** [comment]
The bug was raised with 'Prefer Tab characters' disabled, with indents set at 2 spaces and tab characters set at 8.

I've since tried setting indents at 12 spaces and get this:

```
def bar():
            if False:
                        print 'way over here'
                print 'after shift-TAB or ctl-c, <'
```

So it seems that Decrease-Indentation operates on the tab-size setting, not the 'spaces per indent' setting. I've updated the title accordingly.

At least that means I can set them both to the same value to continue editing sanely.

This is a regression in Komodo X. Komodo 9 works as expected (using spaces-per-indent when reducing indentation level).
